### ISSUE TITLE
MD052: Reference definitions inside blockquotes not recognized

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#345]
## Problem

Reference link definitions inside blockquotes are not being recognized by the MD052 parser.

## Example

```markdown
> Some text with a [link][ref].
>
> [ref]: https://example.com
```

This produces:
```
error[MD052]: Reference link uses undefined label 'ref'
```

The reference definition `[ref]: https://example.com` is valid markdown inside the blockquote, but the parser only looks for definitions at the start of a line (not after `> ` prefixes).

## Root Cause

The `RefDefParser` in `md052.rs` uses `skip_whitespace()` at the start of each line, then checks for `[` to begin a definition. It doesn't account for blockquote markers (`>`).

## Discovered

Found while analyzing issues reported by the rust-embedded/discovery-mb2 project (rust-embedded/discovery-mb2#88).
